common/src/hash.rs

```diff
@@ -86,6 +86,11 @@ pub fn hash_pointer(value: usize) -> PyHash {
     hash as _
 }
 
+#[inline]
+pub fn hash_integer<T: num_traits::PrimInt>(data: T) -> PyHash {
```

Comment:
To avoid unnecessarily complicated error handlings, `pub fn hash_usize(data: usize)` will be enough.
Let's place this function beside `hash_bigint` to find each other easier